LLama.Examples/Examples/SpeechTranscription.cs

```diff
@@ -0,0 +1,201 @@
+        {
+            bool IsOfInterest(string AudioTranscription);
+            void ProcessText(string AudioTranscription);
+        }
+
+        public class SpeechRecognitionServer : IDisposable
+        {
+            const int clipLength = 250; // ms
+            const float voiceDetectionThreshold = 0.01f; // Adjust as needed
+            readonly string[] knownFalsePositives = ["[BLANK_AUDIO]", "Thank you", "[silence]"];
+
+            WaveInEvent waveIn;
+            WaveFormat waveFormat = new(16000, 16, 1); // 16KHz, 16 bits, Mono Channel
+            List<byte> recordedBytes = [];
+
+            WhisperFactory? whisperFactory;
+            WhisperProcessor? processor;
+            string whisperPrompt =
+"""
+The short audio comes from a user that is speaking to an AI Language Model in real time.
+Pay extra attentions for commands like 'ok stop' or just 'stop'.
+In case of inaudible sentences that might be, assume they're saying 'stop'.
+""".Trim();
+
+            // Tracked stats for Speech Recognition, Parsing, and Serving.
+            int currentBlankClips;  // Ideally would work with milliseconds,
+            int totalNonBlankClips; // ..but for example's sake they work on a
+            int nonIdleTime;        // ..clip-based quant-length (1 = clipLength).
+            public (int minBlanksPerSeperation, int minNonBlanksForValidMessages) detectionSettings = (2, 3);
+            // Default detection settings: A speech of 750ms, followed by pause of 500ms. (2x250ms)
+
+            public HashSet<ISpeechRecognitionServiceUser> ServiceUsers = [];
+
+            public SpeechRecognitionServer(string modelPath)
+            {
+                // Adjust the path based on your GPU's type. On your build you ideally want just the correct runtime build for your project, but here we're having all references, so it's getting confused.
+                var libPath = @$"{Environment.GetFolderPath(Environment.SpecialFolder.UserProfile)}\.nuget\packages\whisper.net.runtime.cublas\1.5.0\build\win-x64\whisper.dll"; // Defaulting to cuBlas.
+                if (!File.Exists(libPath)) { Console.Error.WriteLine($"Could not find dll file at {libPath}.\nWhisper will load with the default runtime (possibly CPU)."); libPath = null; }
+                whisperFactory = WhisperFactory.FromPath(modelPath, libraryPath: libPath);
+
+                var builder = whisperFactory.CreateBuilder().WithThreads(16).WithPrompt(whisperPrompt).WithSingleSegment().WithLanguage("en");
+                (builder.WithBeamSearchSamplingStrategy() as BeamSearchSamplingStrategyBuilder)!.WithPatience(0.2f).WithBeamSize(5);
+                processor = builder.Build();
+
+                waveIn = new WaveInEvent() { BufferMilliseconds = clipLength, WaveFormat = waveFormat };
+                waveIn.DataAvailable += WaveIn_DataAvailable;
+                waveIn.StartRecording();
+            }
+
+            void WaveIn_DataAvailable(object? sender, WaveInEventArgs e)
```

Comment:
Please keep the naming convention the same with others here (WaveInDataAvailable).